src/sources/docker.rs

```diff
@@ -263,97 +283,170 @@ impl DockerSource {
+            filters.insert("label".to_owned(), include_labels.clone());
+        }
 
-        // Future
         self.esb
             .core
             .docker
-            .containers()
-            .list(&options.build())
-            .map(move |list| {
-                for container in list {
-                    trace!(
-                        message = "Found already running container",
-                        id = field::display(&container.id),
-                        names = field::debug(&container.names)
-                    );
-
-                    if !self.exclude_vector(container.id.as_str(), container.image.as_str()) {
-                        continue;
-                    }
+            .list_containers(Some(ListContainersOptions {
+                all: false, // only running containers
+                filters,
+                ..Default::default()
+            }))
+            .await?
+            .into_iter()
+            .for_each(|container| {
+                let id = container.id.unwrap();
+                let names = container.names.unwrap();
+                let image = container.image.unwrap();
+
+                trace!(
+                    message = "Found already running container",
+                    id = field::display(&id),
+                    names = field::debug(&names)
+                );
 
-                    // This check is necessary since shiplift doesn't have way to include
-                    // names into request to docker.
-                    if !self.esb.core.config.container_name_included(
-                        container.id.as_str(),
-                        container.names.iter().map(|s| {
-                            // In this case shiplift gives names with starting '/' so it needs to be removed.
-                            let s = s.as_str();
-                            if s.starts_with('/') {
-                                s.split_at('/'.len_utf8()).1
-                            } else {
-                                s
-                            }
-                        }),
-                    ) {
-                        trace!(
-                            message = "Container excluded",
-                            id = field::display(&container.id)
-                        );
-                        continue;
-                    }
+                if !self.exclude_vector(id.as_str(), image.as_str()) {
+                    return;
+                }
 
-                    // Include image check
-                    if let Some(images) = self.esb.core.config.include_images.as_ref() {
-                        let image_check = images.iter().any(|image| &container.image == image);
-                        if !images.is_empty() && !image_check {
-                            continue;
+                // This check is necessary since docker api doesn't have way to include
+                // names into request.
```

Comment:
you can actually filter on names and image id's in the container inspect request by setting the `name` and `ancestor` fields to a vector in the `filters` hashmap.

Reply:
Thanks! I added `ancestor` filter, looks good. Comment about container name is not correct actually, this option works not like match instead `starts_with` applied. I removed wrong comment.